packages/node_modules/@ciscospark/spark-core/src/lib/credentials/credentials.js

```diff
@@ -350,14 +349,14 @@ const Credentials = SparkPlugin.extend({
       this.logger.warn('credentials: failed to clear supertoken', err);
     }
 
-    tokens.forEach((token) => {
+    while (this.userTokens.models.length) {
       try {
-        this.userTokens.remove(token);
+        this.userTokens.remove(this.userTokens.models[0]);
```

Comment:
is there a case where userTokens.remove fails? would it make sense to clear all the tokens using `this.userTokens.reset()`?

Reply:
@aimeex3 yea, Ian said that there was a reason why he looped it initially but he forgot. And I didn't want to find out why so :P

Reply:
it may have something to do with the way the storage layer is rigged up. I don't remember for sure, but it's possible that `reset()` doesn't cause the storage layer to remove stuff.